File: server/src/main/java/bit/project/server/entity/Customerreturn.java
package bit.project.server.entity;

import lombok.Data;
import java.util.List;
import javax.persistence.*;
import java.time.LocalDate;
import javax.persistence.Id;
import java.math.BigDecimal;
import javax.persistence.Lob;
import java.time.LocalDateTime;
import lombok.NoArgsConstructor;
import javax.persistence.Entity;
import javax.persistence.GeneratedValue;
import javax.persistence.GenerationType;
import com.fasterxml.jackson.annotation.JsonInclude;
import com.fasterxml.jackson.annotation.JsonIgnoreProperties;

@Data
@Entity
@NoArgsConstructor
@JsonInclude(JsonInclude.Include.NON_NULL)
@JsonIgnoreProperties({"hibernateLazyInitializer", "handler"})
public class Customerreturn{
    @Id
    @GeneratedValue(strategy=GenerationType.IDENTITY)
    private Integer id;

    private String code;

    private LocalDate date;

    @Lob
    private String reason;

    private BigDecimal amount;

    private String chequeno;

    private LocalDate chequedate;

    private String chequebank;

    private String chequebranch;

    @Lob
    private String description;

    private LocalDateTime tocreation;


    @ManyToOne
    private Sale sale;

    @ManyToOne
    private Paymenttype paymenttype;

    @ManyToOne
    private Paymentstatus paymentstatus;

    @OneToMany(mappedBy="customerreturn", cascade = CascadeType.ALL, orphanRemoval = true)
    private List<Customerreturnitem> customerreturnitemList;

    @ManyToOne
    @JsonIgnoreProperties({"creator","status","tocreation","roleList"})
    private User creator;


    public Customerreturn(Integer id){
        this.id = id;
    }

    public Customerreturn(Integer id, String code, Sale sale, LocalDate date, BigDecimal amount){
        this.id = id;
        this.code = code;
        this.sale = sale;
        this.date = date;
        this.amount = amount;
    }

}

File: client/src/app/entities/customerreturn.ts
import {Sale} from './sale';
import {User} from './user';
import {Paymenttype} from './paymenttype';
import {DataPage} from '../shared/data-page';
import {Paymentstatus} from './paymentstatus';
import {Customerreturnitem} from './customerreturnitem';

export class Customerreturn {
  id: number;
  code: string;
  sale: Sale;
  date: string;
  reason: string;
  amount: number;
  paymenttype: Paymenttype;
  paymentstatus: Paymentstatus;
  chequeno: string;
  chequedate: string;
  chequebank: string;
  chequebranch: string;
  description: string;
  customerreturnitemList: Customerreturnitem[];
  creator: User;
  tocreation: string;
}

export class CustomerreturnDataPage extends DataPage{
    content: Customerreturn[];
}
